inc/composer/class-command.php

```diff
@@ -654,6 +661,179 @@ protected function db( InputInterface $input, OutputInterface $output ) {
+				$os = php_uname( 's' ); # 'Darwin', 'Linux', 'Windows'
+				$arch = php_uname( 'm' ); # 'arm64' for arm, 'x86_64' or 'amd64' for x64
+				$mkcert_version = 'v1.4.3';
+
+				switch( $os ) {
+					# macOS
+					case 'Darwin':
+						$binary_arch = $arch === 'x86_64' ? 'darwin-amd64' : 'darwin-arm64';
+						break;
+					# Linux
+					case 'Linux':
+						$binary_arch = $arch === 'amd64' ? 'linux-amd64' : 'linux-arm64';
+						break;
+					# Windows
+					case 'Windows':
+						$binary_arch = 'windows-amd64.exe';
+						break;
+					default:
+						$binary_arch = null;
+						break;
+				}
+
+				// If couldn't detect a support architecture, ask the user to install mkcert manually.
+				if ( ! $binary_arch ) {
+					$output->writeln( '<error>This command is only supported on macOS, Linux, and Windows x64, install `mkcert` manually for other systems.</error>' );
+					return 1;
+				}
+
+				$binary = "mkcert-$mkcert_version-$binary_arch";
+				$mkcert = "vendor/mkcert";
+
+				// Check if mkcert is installed globally already, bail if so.
+				$version = trim( shell_exec( 'mkcert -version' ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed globally already</error>" );
+					return 1;
+				}
+
+				// Check if mkcert is installed locally already, bail if so.
+				$version = trim( shell_exec( "$mkcert -version" ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed locally already</error>" );
+					return 1;
+				}
+
+				exec( "curl -o $mkcert -L https://github.com/FiloSottile/mkcert/releases/download/$mkcert_version/$binary", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not download mkcert binary, try using sudo or manually installing mkcert.</error>" );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert $mkcert_version was downloaded.</info>" );
+
+				chmod( $mkcert, 0755);
+
+				exec( "$mkcert -version", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not launch mkcert binary, try manually installing mkcert.</error>" );
+					return 1;
+				}
+				$output->writeln( "<info>mkcert $mkcert_version was installed.</info>" );
+
+				// Setup and accept the root certificate.
+				exec( "$mkcert -install", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not setup mkcert properly, try manually installing mkcert.</error>" );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert root CA was installed and accepted successfully.</info>" );
+				return 0;
+				break;
+			case 'generate':
+				$mkcert = $this->get_mkcert_binary();
+				if ( ! $mkcert ) {
+					$output->writeln( "<error>mkcert is not installed, run 'composer server ssl install' or install mkcert manually first.</error>" );
+					return 1;
+				}
+
+				// TODO figure out how to programmatically detect the domains to use
+				$domains = $input->getArgument( 'options' )[1] ?? '*.altis.dev';
```

Comment:
Can I provide a space separated list of domains still?

Reply:
Regarding programmatically detecting the domains to use let's just use the `composer.json` config, maybe we can have the `tld` option accept an array of TLDs and we generate certs for each by default. Would need to merge that configured list with whatever might be passed to this command.

Reply:
That's how it works now with the updated PR that introduced the config detection, however, it does not merge with the config, assuming the user knows what they're doing. Can be an improvement though.

Reply:
And yes you can specify multiple domains, including wildcard domains on the command line.